src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -366,6 +372,28 @@ def _build_kwargs(self, func, ns):  # pylint: disable=no-self-use
             kwargs['ns'] = ns
         return kwargs
 
+    def _rudimentary_get_command(self, args):  # pylint: disable=no-self-use
+        """ Rudimentary parsing to get the command """
+        nouns = []
+
+        for arg in args:
+            if arg[0] != '-':
+                nouns.append(arg)
+            else:
+                break
+
+        def _find_args(args):
+            search = ' '.join(args)
+            try:
+                return self.commands_loader.command_table[search]
+            except KeyError:
+                return False
+
+        # since the command name may be immediately followed by a positional arg, strip those off
```

Comment:
Can't this positional argument at random places?

Reply:
It can, but then the normal method of finding the first thing that starts with `-` is fine.  The only time it causes a problem is when the command is *immediately* followed by the positional arg.

Reply:
good.